internal/ingestion/component/vision_enhancement.go

```
@@ -0,0 +1,275 @@
+
+	// 2. Resolve tenant's IMAGE2TEXT model.
+	driver, modelName, apiConfig, _, err := resolveTenantModelByType(ctx, db, tenantID, entity.ModelTypeImage2Text)
+	if err != nil {
+		// Model not available — skip vision enhancement silently, matching Python's try/except pass.
+		return dispatched, false, nil
+	}
+
+	// Hoist prompt once: language is invariant across items.
+	prompt, perr := figureVisionPromptBuilder(language)
+	if perr != nil {
+		return dispatched, false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 140) but it returns nil

(nilerr)